cmd/to-crdgen/main.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2019. PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	crdutils "github.com/ant31/crd-validation/pkg"
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap.com/v1alpha1"
+	k8sutil "github.com/pingcap/tidb-operator/pkg/util"
+	extensionsobj "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"os"
+)
+
+var (
+	cfg crdutils.Config
+)
+
+func initFlags(crdkind v1alpha1.CrdKind, flagset *flag.FlagSet) *flag.FlagSet {
+	flagset.Var(&cfg.Labels, "labels", "Labels")
+	flagset.Var(&cfg.Annotations, "annotations", "Annotations")
+	flagset.BoolVar(&cfg.EnableValidation, "with-validation", true, "Add CRD validation field, default: true")
+	flagset.StringVar(&cfg.Group, "apigroup", v1alpha1.GroupName, "CRD api group")
+	flagset.StringVar(&cfg.SpecDefinitionName, "spec-name", crdkind.SpecName, "CRD spec definition name")
+	flagset.StringVar(&cfg.OutputFormat, "output", "yaml", "output format: json|yaml")
+	flagset.StringVar(&cfg.Kind, "kind", crdkind.Kind, "CRD Kind")
+	flagset.StringVar(&cfg.ResourceScope, "scope", string(extensionsobj.NamespaceScoped), "CRD scope: 'Namespaced' | 'Cluster'.  Default: Namespaced")
+	flagset.StringVar(&cfg.Version, "version", v1alpha1.Version, "CRD version, default: 'v1alpha1'")
+	flagset.StringVar(&cfg.Plural, "plural", crdkind.Plural, "CRD plural name")
+	return flagset
+}
+
+func init() {
+	var command *flag.FlagSet
+	if len(os.Args) == 1 {
```

Comment:
Maybe you can use a library to parse the command line arguments?

Reply:
fix